crypto/relic_build.sh

```diff
@@ -34,21 +39,61 @@ EP_METH=(-DEP_MIXED=ON -DEP_PLAIN=OFF -DEP_SUPER=OFF -DEP_DEPTH=4 -DEP_WIDTH=2 \
     -DEP_CTMAP=ON -DEP_METHD="JACOB;LWNAF;COMBS;INTER")
 PP_METH=(-DPP_METHD="LAZYR;OATEP")
 
-# Generate make files using cmake
-cmake "${COMP[@]}" "${GENERAL[@]}" \
+# "-march=native" is not supported on ARM by clang
+# we probe the cmake compiler detection logic by running cmake once with
+# the permissive logic, and adding the '-march=native' option only if
+# supported
+PROBE_COMP=(-DCOMP="-O3 -funroll-loops -fomit-frame-pointer -mtune=native")
+
+# make cmake print its CC interpretation
+echo 'message ( STATUS "CC=$ENV{CC}" )' >> ../CMakeLists.txt
+
+# Probe cmake's MakeFile generation
+CMAKE_OUTPUT=$(mktemp)
+cmake "${PROBE_COMP[@]}" "${GENERAL[@]}" \
     "${LIBS[@]}" "${RAND[@]}" \
     "${BN_REP[@]}" "${ARITH[@]}" \
     "${PRIME[@]}" "${PRIMES[@]}" \
     "${EP_METH[@]}" \
     "${BN_METH[@]}" \
     "${FP_METH[@]}" \
     "${FPX_METH[@]}" \
-    "${PP_METH[@]}" ..
+    "${PP_METH[@]}" .. > $CMAKE_OUTPUT
+# de-mangle the CMakeLists file, done with a temp file
+# to be portable between GNU / BSD sed
+CMAKE_TEMP=$(mktemp)
+sed -e '/message ( STATUS "CC=$ENV{CC}" )/d' ../CMakeLists.txt > $CMAKE_TEMP
```

Comment:
As the comment above the line you comment on mentions, the `-i` option is not supported on BSD sed, and was therefore not used. See [this comment](https://github.com/onflow/flow-go/pull/823#discussion_r657112980) for what I tested, though that may have been insufficient.

Reply:
> Maybe we could just avoid deleting the extra line? or use grep and only write the extra line if it doesn't exist already?

1. It's important to delete the added line, because the `CMakeLists` file is part of the relic build, which is part of a submodule. The change made here could end up in a (flow) commit very easily.
2. If we add the line only if it's not here already, the `tail` logic we use to grab the `CC=...` line may fail (because a CC print may occur only earlier in the output), which may silently cause a divergence between cmake's notion of CC and that of the present script.

Reply:
I agree deleting the line would be cleaner, I just wanted to avoid using `sed`. 
The `sed` command to delete the line works fine on my machine 👌